driver-core/src/main/com/mongodb/connection/SimpleCommandProtocol.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright 2008-2016 MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.connection;
+
+import com.mongodb.MongoNamespace;
+import com.mongodb.ReadPreference;
+import com.mongodb.async.SingleResultCallback;
+import com.mongodb.diagnostics.logging.Logger;
+import com.mongodb.diagnostics.logging.Loggers;
+import org.bson.BsonDocument;
+import org.bson.FieldNameValidator;
+import org.bson.codecs.Decoder;
+
+import static com.mongodb.assertions.Assertions.notNull;
+import static com.mongodb.connection.ProtocolHelper.getMessageSettings;
+import static java.lang.String.format;
+
+/**
+ * A protocol for executing a command against a MongoDB server using the OP_QUERY wire protocol message.
+ *
+ * @param <T> the type returned from execution
+ * @mongodb.driver.manual ../meta-driver/latest/legacy/mongodb-wire-protocol/#op-query OP_QUERY
+ */
+class SimpleCommandProtocol<T> implements CommandProtocol<T> {
```

Comment:
Sorry about this:  It's not a new file really, it's just what used to be in CommandProtocol, with some small changes to propagate the session context.